src/main/java/com/sap/cloudfoundry/client/facade/rest/CloudControllerRestClientImpl.java

```diff
@@ -2587,21 +2595,17 @@ private UUID getOrganizationGuid(String organizationName, boolean required) {
             : null;
     }
 
-    private Map<String, UUID> getDomainGuids() {
-        List<CloudDomain> availableDomains = new ArrayList<>();
-        availableDomains.addAll(getDomainsForOrganization());
-        availableDomains.addAll(getSharedDomains());
-        Map<String, UUID> domains = new HashMap<>(availableDomains.size());
-        for (CloudDomain availableDomain : availableDomains) {
-            domains.put(availableDomain.getName(), availableDomain.getMetadata()
-                                                                  .getGuid());
-        }
-        return domains;
+    private Map<String, UUID> getDomainsFromRoutes(Set<CloudRouteSummary> routes) {
+        List<String> domainNames = routes.stream()
+                                         .map(CloudRouteSummary::getDomain)
+                                         .filter(domain -> !StringUtils.isEmpty(domain))
```

Comment:
In which case domain can be null or empty?

Reply:
It should never be. But the `CloudRouteSummary` objects that are from the `CloudApplication` object are parsed from a /v2/summary response. The null/empty check is just in case the v2 response is incomplete/incorrect.